Gems/AWSClientAuth/Code/Tests/AWSClientAuthGemMock.h

```diff
@@ -542,7 +542,7 @@ namespace AWSClientAuthUnitTest
             m_jobContext.reset(aznew AZ::JobContext(*m_jobManager, *m_jobCancelGroup));
             AZ::JobContext::SetGlobalContext(m_jobContext.get());
 
-            AWSNativeSDKInit::InitializationManager::InitAwsApi();
+            AWSNativeSDKInit::InitializationManager::InitAwsApiNoLogging();
```

Comment:
Seem dangerous to suppress logging on all tests. This could cause us to miss genuine cases where the underlying code is making network calls in unit tests.

It feels like we are just suppressing all logs rather than dealing with correctly setting up fixtures, mocks etc so we are avoiding the underlying systems.

Reply:
The problem is any practical use case that we need to make aws http request call in unit tests? if not, then why do we care about the underlying logging?

And more importantly, gem unit testing should not care sdk logging, that is the purpose of gem unit test which we just test the functionality of gem unit.

Reply:
if you ask me what is optimal solution, I would say inject mock AWS Config constructor (I don't know how to do that actually, but willing to take any suggestion)
besides that, this one is better and easier. As if we suppress error, or create env variable or create temp config file, they are either doing the same thing, or not aligned with unit test purpose.

Reply:
how difficult would it be to set `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` values in the test environment? This will satisfy the AWS Config before it attempts to look for a config profile or hit the EC2 metadata endpoint

Reply:
TBH, I don't know how hard it could be.
But to me, it is still not an optimal solution, as we try to provide env variable because we can't mock AWS Config.